### PR TITLE
Feature/kotlin build

### PR DESCRIPTION
**Pull request by erikeelde**
Move to kotlin gradle build scripts. Only migration - no changes. 
Ran into some lint problems that needed to be fixed.

### CODE REVIEW

**Review by joezorry — APPROVED** [review]
Awesome. Should do this too.
Did you stumble upon any problems?

**Comment by erikeelde** [comment]
Actually not. The first file converted I did make an 'invalidate caches and restart' for it to work. Before that everything was missing symbols. Then it was just a simple matter of finding where things existed in the kotlin APIs  - fairly easy really. I accidentally used File(path: String) rather than file(paht: String) in some places which rendered the lint-engine to crash a few times. 

Outside of that feels pretty polished and the autocomplete you get is really fantastic. So feels like a win this far. I have yet to try out the ben manes publishing scripts though - but know for a fact it's finding it properly (as it crashed when I configured it wrongly) - kinda left them out as I'm contemplating the plugin that extends the publish scripts anyways so kinda wanted to pick paths there before investing time with it.

**Comment by joezorry** [comment]
Really nice. Getting me psyched to try it out myself!

**Review by tingnapianhai — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+buildscript {
+    Repos.addRepos(repositories)
+
+    dependencies {
+
+        classpath(Libs.android_gradle_plugin)
+        classpath(Libs.kotlin.plugin)
+        classpath(Libs.navigation.safeArgsPlugin)
+        classpath(Libs.oss.plugin)
+        classpath(Libs.kotlin.plugin)
```
Comment (joezorry) [inline build.gradle.kts]: It looks like this is duplicated within this scope.

Comment (erikeelde) [reply build.gradle.kts]: 👍

**Review by joezorry — APPROVED** [review]
Loooooking good.

**Pull request merged**